src/modules/MouseUtils/MouseHighlighter/dllmain.cpp

```diff
@@ -0,0 +1,316 @@
+        if (settingsObject.GetView().Size())
+        {
+            try
+            {
+                // Parse HotKey
+                auto jsonPropertiesObject = settingsObject.GetNamedObject(JSON_KEY_PROPERTIES).GetNamedObject(JSON_KEY_ACTIVATION_SHORTCUT);
+                auto hotkey = PowerToysSettings::HotkeyObject::from_json(jsonPropertiesObject);
+                m_hotkey = HotkeyEx();
+                if (hotkey.win_pressed())
+                {
+                    m_hotkey.modifiersMask |= MOD_WIN;
+                }
+
+                if (hotkey.ctrl_pressed())
+                {
+                    m_hotkey.modifiersMask |= MOD_CONTROL;
+                }
+
+                if (hotkey.shift_pressed())
+                {
+                    m_hotkey.modifiersMask |= MOD_SHIFT;
+                }
+
+                if (hotkey.alt_pressed())
+                {
+                    m_hotkey.modifiersMask |= MOD_ALT;
+                }
+
+                m_hotkey.vkCode = hotkey.get_code();
+            }
+            catch (...)
+            {
+                Logger::warn("Failed to initialize Mouse Highlighter activation shortcut");
+            }
+            uint8_t opacity = MOUSE_HIGHLIGHTER_DEFAULT_OPACITY;
+            try
+            {
+                // Parse Opacity
+                auto jsonPropertiesObject = settingsObject.GetNamedObject(JSON_KEY_PROPERTIES).GetNamedObject(JSON_KEY_HIGHLIGHT_OPACITY);
+                opacity = (uint8_t)jsonPropertiesObject.GetNamedNumber(JSON_KEY_VALUE);
+            }
+            catch (...)
+            {
+                Logger::warn("Failed to initialize Opacity from settings. Will use default value");
+            }
+            try
+            {
+                // Parse left button click color
+                auto jsonPropertiesObject = settingsObject.GetNamedObject(JSON_KEY_PROPERTIES).GetNamedObject(JSON_KEY_LEFT_BUTTON_CLICK_COLOR);
+                auto leftColor = (std::wstring)jsonPropertiesObject.GetNamedString(JSON_KEY_VALUE);
+                uint8_t r, g, b;
+                if (!checkValidRGB(leftColor,&r,&g,&b))
+                {
+                    throw 1;
+                }
+                highlightSettings.leftButtonColor = winrt::Windows::UI::ColorHelper::FromArgb(opacity, r, g, b);
+            }
+            catch (...)
+            {
+                Logger::warn("Failed to initialize left click color from settings. Will use default value");
+            }
+            try
+            {
+                // Parse right button click color
+                auto jsonPropertiesObject = settingsObject.GetNamedObject(JSON_KEY_PROPERTIES).GetNamedObject(JSON_KEY_RIGHT_BUTTON_CLICK_COLOR);
+                auto leftColor = (std::wstring)jsonPropertiesObject.GetNamedString(JSON_KEY_VALUE);
+                uint8_t r, g, b;
+                if (!checkValidRGB(leftColor, &r, &g, &b))
+                {
+                    throw 1;
```

Comment:
I think it would be better to return an error code instead of throwing a magic number, it would be more readable.

Reply:
I think this is fine, just throwing to jump to logging in catch block

Reply:
It's an actual exception in this case, since it'd probably mean manually manipulated config file or non-existing config field.
Otherwise, it'd mean duplicating log code.
Might had a comment there to make it clearer.

Reply:
Perhaps a different log error can be added. Will do it in a `if`, as suggested.